### PR TITLE
Add replace filter

### PR DESCRIPTION
**Pull request by SmilingNavern**
Imitating simple replacement without regexps

### CODE REVIEW

**Comment by SmilingNavern** [comment]
Thanks for feedback i am going to re-read code and fix it accordingly.

**Comment by SmilingNavern** [comment]
@bitfield 
I have fixed code as you requested. I am not good at english so i have used your descriptions for Readme and comments for Replace functions.

**Comment by SmilingNavern** [comment]
Use case is simple it is really often that you need to perform sed or another type of substitute for text or for config files. So Replace function is imitating simple sed in a way

**Comment by bitfield** [comment]
> Use case is simple it is really often that you need to perform sed or another type of substitute for text or for config files

Yes, I understand that; but presumably you wanted to use `script` to write a specific program to do something like this. If you could give me a simple version of that program it would be very helpful.

File: filters.go [hunk 1]
```diff
@@ -34,7 +34,9 @@ func (p *Pipe) MatchRegexp(re *regexp.Regexp) *Pipe {
 	})
 }
 
-// Replace reads from the pipe, and returns a new pipe containing replaces lines
+// Replace reads from the pipe, and returns a new pipe
```
Comment (bitfield) [inline filters.go]: Could you wrap these comments to 80 columns, for consistency with other comments? I use the Rewrap plugin in VS Code for this, which saves a lot of manual reformatting.

Comment (SmilingNavern) [reply filters.go]: Yep, okay:)

**Comment by bitfield** [comment]
Hey @SmilingNavern—are you still interested in working on this? If not, that's fine, but would you mind if I picked up your PR and finished off the changes requested here?

**Comment by SmilingNavern** [comment]
Sorry, i am still interested and i will return to this a bit later but if you in hurry feel free to finish this)

**Comment by bitfield** [comment]
@SmilingNavern I've rebased this and added a corresponding `ReplaceRegexp()` to complement yours. If you're happy, I'm happy!

@breml would you take a look at this and let me know what you think?

File: filters.go [hunk 2]
```diff
@@ -193,6 +193,7 @@ func (p *Pipe) MatchRegexp(re *regexp.Regexp) *Pipe {
 	})
 }
 
+
```
[the remark this answers is not in the document]
Comment (bitfield) [reply filters.go]: It is `gofmt`ed, but there's an extra bogus blank line in there. Nice catch!

File: filters_test.go [hunk 3]
```diff
@@ -95,6 +95,54 @@ func TestMatchRegexp(t *testing.T) {
...
+	t.Parallel()
+	testCases := []struct {
+		testFileName string
+		search       string
+		replace      string
+		want         string
+	}{
+		{"testdata/hello.txt", "hello", "bye", "bye world\n"},
+		{"testdata/hello.txt", "Does not exist in input", "Will not appear in output", "hello world\n"},
+		{"testdata/hello.txt", " world", " string with newline", "hello string with newline\n"},
```
[the remark this answers is not in the document]
Comment (bitfield) [reply filters_test.go]: Well spotted! I took out the newline...

**Comment by bitfield** [comment]
Thanks @breml—great job.

**Comment by bitfield** [comment]
Woohoo @SmilingNavern—your PR is officially _in_! Nice work, thank you!

**Pull request merged**